File: backend/springnatour/src/main/java/com/unina/springnatour/dto/user/UserDto.java
package com.unina.springnatour.dto.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.io.Serializable;

@Data
public class UserDto implements Serializable {
    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("username")
    private String username;

    @JsonProperty("profile_photo")
    private String profilePhoto;
}

File: client/app/src/main/java/com/unina/natourkt/core/data/remote/dto/UserDto.kt
package com.unina.natourkt.core.data.remote.dto

import com.google.gson.annotations.SerializedName
import com.unina.natourkt.core.domain.model.User

/**
 * Response coming from API for [User]
 */
data class UserDto(
    @SerializedName("user_id")
    val id: Long,

    @SerializedName("username")
    val username: String,

    @SerializedName("profile_photo")
    val profilePhoto: String?,
)